rskj-core/src/main/java/co/rsk/rpc/netty/JsonRpcWeb3ServerHandler.java

```diff
@@ -58,10 +70,21 @@ protected void channelRead0(ChannelHandlerContext ctx, ByteBufHolder request) th
         ByteBuf responseContent = Unpooled.buffer();
         int responseCode;
         try (ByteBufOutputStream os = new ByteBufOutputStream(responseContent);
-             ByteBufInputStream is = new ByteBufInputStream(request.content().retain())){
-            
+             ByteBufInputStream is = new ByteBufInputStream(request.content().retain())) {
+
             responseCode = jsonRpcServer.handleRequest(is, os);
-        } catch (Exception e) {
+        } catch (JsonRpcRequestPayloadException e) {
+            String invalidReqMsg = "Invalid request";
+            LOGGER.error(invalidReqMsg, e);
+            responseContent = buildErrorContent(ErrorResolver.JsonError.INVALID_REQUEST.code, e.getMessage());
+            responseCode = ErrorResolver.JsonError.INVALID_REQUEST.code;
+        } catch (StackOverflowError e) {
+            String stackOverflowErrorMsg = "Stack overflow produced due to complex nested objects detected in request params";
```

Comment:
```suggestion
            String stackOverflowErrorMsg = "Invalid request";
```
I'd try to be a bit concise here, as this message is being sent to the end user